### DISCUSSION TITLE
Has anyone else had their suggestions become way worse lately?

### DISCUSSION THREAD

**Discussion by colecrouter** [community/community#12852] [Copilot Conversations]
I've been using Copilot since the beta launch for programming in Go, and it's been awesome to use. It has solved almost everything I've thrown at it, including builtin packages, functions/variables exclusive to a workspace, and everything else. I took a break and got back recently, and I've noticed that Copilot's suggestions seem completely broken. 

I had `// Create a temp folder` in on of my projects, which previously generated this (back in November?):
```go
// Create a temp folder
tmp, err := os.MkdirTemp("", "test")
if err != nil {
	panic(err)
}
defer os.RemoveAll(tmp)
```
Which is bang on. Going back to the same file, at the same commit, I get this now:
```go
// Create a temp folder
tempFolder = createTempFolder()
```
or
```go
// Create a temp folder
tmp, err := CreateTempFolder()
```
> It's worth mentioning that neither of these functions, nor `tempFolder` exists anywhere in the project.

I used to be able to write whole unit tests just by hitting tab, now I can't seem to get anything other than comments using the wrong variable/function names.

Has anyone else experienced anything similar recently? I haven't had a chance to test other languages.

**Comment by tmlewin** [comment]
> ed any

I also noticed it as well. The quality of output and results has dropped. i dont know if it still on gpt 3? It doesnt handle comments as well as it used to be.

**Comment by hackf5** [comment]
Yes, the whole team has been commenting on it over the past week or so. In C# it just seems to spew random gibberish at the moment. Absolutely no contextual knowledge whatsoever. Using libraries that have never been used in the project, etc...

**Comment by Skirlax** [comment]
Yes same for me. In Python, Java , Dart. It got worse across the board.

**Comment by Jameswood-vega** [comment]
same here!

**Comment by AlliedToasters** [comment]
Same here:
python, javascript, sql.

I'm gonna cancel my sub.

**Comment by ghost** [comment]
Yes, it's horrible this days, I almost don't user it's suggestions anymore, thinking to switch to other solution.

**Comment by EtienneDavid** [comment]
Same here ! It starts to be completly random, especially with the jupyter notebook !

**Comment by baldoOsu** [comment]
Same, used it in beta and it was awesome. Just tried it out again and it's terrible.

**Comment by LukeGahan** [comment]
Yes same in python using notebooks, in my case it no longer seems to be aware of code outside of the current cell (which I accept could be a vscode issue)

It's pretty much useless for me at the moment... The experience used to be much better

**Comment by pgrisins** [comment]
Same here, useless suggestions. A few months ago it was simply amazing

**Comment by bala-srm** [comment]
Same. Massive drop in quality of suggestions and completions

**Comment by bash-j** [comment]
Same. Not working in jupyter notebook files. The copilot chat seemed promising, but I don't think I have gotten a single useful answer out of it.

**Comment by alpianon** [comment]
it's strange, because I started using copilot for a specific project 2 months ago, and initially it was outstanding, 80% / 90% of suggestions were correct and some were even very smart; but about a couple of weeks ago it started suggesting wrong variable names, use of functions that do not exist, wrong function arguments etc.  so that only 10%-20% of suggestions are actually correct, while the others are not (even if they may seem correct if you don't pay much attention, so I have to spend a lot of time in debugging). I'm still working on the same project: it seems the more I develop such project, the more Copilot isn't able to provide meaningful suggestions.

**Comment by mfirthOmx** [comment]
Such a shame: because it started off great and then became complete garbage. It very simply does nothing more than waste my teams time at this point. Your guys blew it...
